clang/lib/Sema/SemaFunctionEffects.cpp

```diff
@@ -1048,15 +1048,14 @@ class Analyzer {
     }
 
     void checkIndirectCall(CallExpr *Call, QualType CalleeType) {
-      auto *FPT =
-          CalleeType->getAs<FunctionProtoType>(); // Null if FunctionType.
       FunctionEffectKindSet CalleeEffects;
-      if (FPT)
-        CalleeEffects.insert(FPT->getFunctionEffects());
+      if (FunctionEffectsRef Effects = FunctionEffectsRef::get(CalleeType);
```

Comment:
`FunctionEffectsRef::get()` knows how to pick through indirections (references/pointers) to get to a `FunctionProtoType` so just reuse that.